wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentmentFlow.kt

```diff
@@ -136,20 +137,35 @@ suspend fun showPresentmentFlow(
 
                 // show Passphrase prompt
                 is SoftwareSecureArea -> {
+                    // enforce a maximum number of attempts
+                    if (remainingPassphraseAttempts == 0) {
+                        throw IllegalStateException("Error! Reached maximum number of Passphrase attempts.")
+                    }
+                    remainingPassphraseAttempts--
+
                     val softwareKeyInfo =
                         mdocCredential.secureArea.getKeyInfo(mdocCredential.alias) as SoftwareKeyInfo
-
+                    val constraints = softwareKeyInfo.passphraseConstraints!!
+                    val title =
+                        if (constraints.requireNumerical)
+                            activity.resources.getString(R.string.passphrase_prompt_pin_title)
+                        else
+                            activity.resources.getString(R.string.passphrase_prompt_passphrase_title)
+                    val content =
+                        if (constraints.requireNumerical) activity.resources.getString(R.string.passphrase_prompt_pin_content) else activity.resources.getString(
+                            R.string.passphrase_prompt_passphrase_content
+                        )
                     val passphrase = showPassphrasePrompt(
                         activity = activity,
-                        constraints = softwareKeyInfo.passphraseConstraints!!,
-                        checkWeakPassphrase = softwareKeyInfo.isPassphraseProtected,
+                        constraints = constraints,
+                        title = title,
+                        content = content,
                     )
-                    // ensure the passphrase is not empty, else throw IllegalStateException
-                    check(passphrase.isNotEmpty()) { "[Passphrase Unsuccessful]" }
-                    // use the passphrase that the user entered to create the KeyUnlockData
+                    // user cancelled the prompt or no text was provided
+                    check(!passphrase.isNullOrEmpty()) { "[Passphrase Unsuccessful]" }
```

Comment:
What's up with the brackets in error messages? Please remove those (here and elsewhere).